pocs/vesting-flow.html

```diff
@@ -0,0 +1,276 @@
+            },
+            then: {
+              when:
+                currentPeriod == numberOfPeriods
+                  ? [employeeWithdrawCase]
+                  : [employeeWithdrawCase, employerCancelCase],
+              timeout: nextVestingDate,
+              timeout_continuation: continuation,
+            },
+          },
+        };
+      };
+
+      return employerDepositsFunds(contractForPeriod(1));
+    }
+
+    const now = new Date().valueOf();
+
+    const periodLength = 1000 * 60 * 2; // two minutes
+    const contractStart = now + 1000 * 60 * 2; // Contract starts one minute from now
+    const employerDepositDeadline = now + 1000 * 60 * 3; // Employer deposit deadline one and a half minutes from now
+    const amount = 300;
+    const periods = 3;
+
+    const vesting_contract_example_1 = vestingContract(
+      periods,
+      periodLength,
+      { currency_symbol: "", token_name: "" },
+      amount,
+      contractStart,
+      employerDepositDeadline,
+      { role_token: "Employee" },
+      { role_token: "Employer" }
+    );
+
+    console.log(vesting_contract_example_1);
+
+    console.log(`Contract defined to start at ${new Date(contractStart)}`);
+    console.log(
+      `Contract defined with employer deadline at ${new Date(
+        employerDepositDeadline
+      )}`
+    );
+
+    debugger;
+
+    const runtimeLifeCycle = await Browser.mkRuntimeLifecycle({
+      runtimeURL: "https://marlowe-runtime-preview-web.scdev.aws.iohkdev.io/",
+      walletName: "nami", // choose one with getAvailableWallets
+    });
+
+    console.log("Employer creates contract.");
+    const contractId = await runtimeLifeCycle.contracts.create({
+      contract: vesting_contract_example_1,
+      roles: {
+        Employer: await runtimeLifeCycle.wallet.getChangeAddress(),
+        Employee: await runtimeLifeCycle.wallet.getChangeAddress(),
+      },
+    });
+
+    console.log("Employer injects initial funds.");
+    await runtimeLifeCycle.contracts.applyInputs(contractId, (next) => {
+      return {
+        inputs: [
+          {
+            input_from_party: { role_token: "Employer" },
+            that_deposits: amount * periods,
+            of_token: { currency_symbol: "", token_name: "" },
+            into_account: { role_token: "Employer" },
+          },
+        ],
+      };
+    });
+
+    const payoutTime1 = contractStart + 1 * periodLength;
+    const payoutTime2 = contractStart + 2 * periodLength;
+    const payoutTime3 = contractStart + 3 * periodLength;
+
+    console.log(`1st payout available on: ${new Date(payoutTime1)}`);
+    console.log(`2nd payout available on: ${new Date(payoutTime2)}`);
+    console.log(`3rd payout available on: ${new Date(payoutTime3)}`);
+
+    setTimeout(() => {
+      console.log("1st payout is available!");
+      console.log("Poking contract...");
+      // Doesn't seem to work:
+      runtimeLifeCycle.contracts.applyInputs(contractId, (next) => ({
+        inputs: [],
+      }));
```

Comment:
I want to advance the contract when the timeout continuation should be ready to proceed but I'm not sure what happens here. Maybe I need to add more margin than 1000ms on this timeout.